### PR TITLE
Upgrade default flink version to 1.20.0

### PR DESCRIPTION
**Pull request by wForget**
# :mag: Description

Upgrade default flink version to 1.20.0

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [X] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
CI workflow should be updated too.

**Comment by wForget** [comment]
> CI workflow should be updated too.

Changed, thanks

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -256,15 +256,15 @@ jobs:
...
           - java: 8
-            flink: '1.17'
+            flink: '1.20'
             flink-archive: '-Dflink.archive.mirror=https://archive.apache.org/dist/flink/flink-1.19.1 -Dflink.archive.name=flink-1.19.1-bin-scala_2.12.tgz'
             comment: 'verify-on-flink-1.19-binary'
           - java: 8
-            flink: '1.17'
+            flink: '1.20'
             flink-archive: '-Dflink.archive.mirror=https://archive.apache.org/dist/flink/flink-1.20.0 -Dflink.archive.name=flink-1.20.0-bin-scala_2.12.tgz'
             comment: 'verify-on-flink-1.20-binary'
```
Comment (pan3793) [inline .github/workflows/master.yml:269]: now we build the engine with flink 1.20, the binary check list should be 1.17, 1.18 and 1.19

Comment (wForget) [reply .github/workflows/master.yml]: fixed

**Review by pan3793 — APPROVED** [review]
LGTM, pending CI

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**